Platform/MeteorlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c

```diff
@@ -594,6 +598,19 @@ UpdateFspConfig (
     FspsConfig->AtomTurboRatioLimitNumCore[Index]          = 0x0;
   }
 
+  // Set VerbTable is disabled by default. Enable it only when specified by config data.
+  if (FeaturesCfgData != NULL && (FeaturesCfgData->Features.HdaVerbTable != 0)){
+      HdaVerbTablePtr = (UINT32 *) AllocateZeroPool (3 * sizeof (UINT32));
+      HdaVerbTableNum = 0;
+    if (HdaVerbTablePtr != NULL) {
+      DEBUG ((DEBUG_INFO, "HdaVerbTableAlc897 init\n"));
+      HdaVerbTablePtr[HdaVerbTableNum++] = (UINT32)(UINTN) &HdaVerbTableAlc897;
+      FspsConfig->PchHdaVerbTablePtr = (UINT32)(UINTN) HdaVerbTablePtr;
+      FspsConfig->PchHdaVerbTableEntryNum = HdaVerbTableNum;
+    } else {
+      DEBUG ((DEBUG_ERROR, "UpdateFspConfig Error: Could not allocate Memory for HdaVerbTable\n"));
+    }
+  }
```

Comment:
why don't set FspsConfig->PchHdaVerbTablePtr directly using (UINT32)(UINTN) &HdaVerbTableAlc897? This way it could avoid memory allocation and simplify the logic.